tests/test_scalars.py

```diff
@@ -78,27 +95,37 @@ def test_period() -> None:
     as4 = int(1)
     as5 = pd.period_range("2012-1-1", periods=10, freq="D")
     as6 = pd.Period("2012-1-1", freq="D")
+    as7 = cast("TimedeltaSeries", pd.Series([pd.Timedelta(days=1)]))
+    as8 = cast("PeriodSeries", pd.Series([as6]))
 
     check(assert_type(p + as0, pd.Period), pd.Period)
     check(assert_type(p + as1, pd.Period), pd.Period)
     check(assert_type(p + as2, pd.Period), pd.Period)
     check(assert_type(p + as3, pd.Period), pd.Period)
     check(assert_type(p + as4, pd.Period), pd.Period)
-
+    check(assert_type(p + p.freq, pd.Period), pd.Period)
+    check(assert_type(p + (p - as5), pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(p + as7, "PeriodSeries"), pd.Series)
+    das8 = cast("TimedeltaSeries", (as8 - as8))
```

Comment:
would really like to avoid using `cast` in testing.  The result of the subtraction should be `TimedeltaSeries` if we did the typing right.

Reply:
This is an issue for `Series`.  I'm trying to stay as local as possible and would like to not dive into Series.